### ISSUE TITLE
Why CrsMatrix::staticcrsgraph_type uses execution_space and not device_type?

### ISSUE THREAD

**Issue by brian-kelley** [kokkos/kokkos-kernels#665]
@ndellingwood In KokkosSparse_CrsMatrix.hpp, the graph type is defined using execution_space, not device_type. Is this intentional? I think it will need to be changed for Tpetra to remove UVM from CrsMatrix/CrsGraph, because when UVM is enabled as the default memory space for Kokkos::Cuda, it's not possible to instantiate KokkosSparse::CrsMatrix for device = ``Device<Cuda, CudaSpace>``.

```
409 #ifdef KOKKOS_ENABLE_DEPRECATED_CODE
410   //! Type of the graph structure of the sparse matrix.
411   typedef Kokkos::StaticCrsGraph<ordinal_type, Kokkos::LayoutLeft, execution_space, size_type, memory_traits> StaticCrsGraphType;
412   //! Type of the graph structure of the sparse matrix - consistent with Kokkos.
413   typedef Kokkos::StaticCrsGraph<ordinal_type, Kokkos::LayoutLeft, execution_space, size_type, memory_traits> staticcrsgraph_type;
414 #else
415   //! Type of the graph structure of the sparse matrix.
416   typedef Kokkos::StaticCrsGraph<ordinal_type, Kokkos::LayoutLeft, execution_space, memory_traits, size_type> StaticCrsGraphType;
417   //! Type of the graph structure of the sparse matrix - consistent with Kokkos.
418   typedef Kokkos::StaticCrsGraph<ordinal_type, Kokkos::LayoutLeft, execution_space, memory_traits, size_type> staticcrsgraph_type;
419 #endif
```

**Comment by ndellingwood** [comment]
> Is this intentional?

@brian-kelley I don't think so, I'm not sure who wrote the original code but I just checked the Kokkos StaticCrsGraph and the third template parameter is intended to be a DeviceType, not specifically and execution space.

**Comment by brian-kelley** [comment]
@ndellingwood OK, I'll change this to a bug and make a PR.

**Comment by brian-kelley** [comment]
Fixed in #666 (ooh, the PR of the beast)
